### PR TITLE
test, refactor: fund tx deterministically in `feature_coinstatindex`

### PR DESCRIPTION
**Pull request by josibake**
While working on some changes to coin selection in a separate branch, I noticed this test was failing. This is because the test expects exact values from `gettxoutsetinfo` to compare against, which relies on coin selection to be deterministic.

If coin selection is changed in any way, this test will fail and need to be updated with new values.

Instead, we can make the test deterministic by filtering to a UTXO of a specific amount as our first input and then using `fundrawtransaction` to grab a second UTXO. This works because the remaining large UTXOs in the wallet are coinbase outputs. Creating a transaction in this way does not affect any of the logic in the test.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Would it be harder to use MiniWallet, which is deterministic by design?

**Comment by josibake** [comment]
> Would it be harder to use MiniWallet, which is deterministic by design?

Good point! Looking through the full test, it _seems_ like you could use `MiniWallet`, but it's definitely a bigger change than the few lines I did here. Selfishly, I'd prefer making the small change now to unblock myself on a bigger PR I'm working on for coin selection, and then come back to this one later and re-write it to use `MiniWallet` in a follow-up PR.

**Comment by josibake** [comment]
force push to fix the commit message

**Review by willcl-ark — APPROVED** [review]
tACK. Although I didn't have @josibake's  branch to test against (where the tests were failing), the test passes with this patch, and making the test more deterministic here is an improvement.

File: test/functional/feature_coinstatsindex.py [hunk 1]
```diff
@@ -147,11 +147,24 @@ def _test_coin_stats_index(self):
             })
             self.block_sanity_check(res5['block_info'])
 
-        # Generate and send a normal tx with two outputs
+        # Generate and send a normal tx with two outputs.
```
[the remark this answers is not in the document]
Comment (josibake) [reply test/functional/feature_coinstatsindex.py:150]: fixed!

**Comment by MarcoFalke** [comment]
See #24605 as alternative

**Comment by josibake** [comment]
closing in favor of https://github.com/bitcoin/bitcoin/pull/24605

**Pull request closed without merging**